pkgs/data/fonts/google-fonts/default.nix

```diff
@@ -31,13 +39,26 @@ stdenvNoCC.mkDerivation {
 
   dontBuild = true;
 
+  # The font files are in the fonts directory and use two naming schemes:
+  # FamilyName-StyleName.ttf and FamilyName[param1,param2,...].ttf
+  # This installs all fonts if fonts is empty and otherwise only
+  # the specified fonts by FamilyName. To do this, it invokes
+  # `find` 2 times for every font, anyone is free to do this
+  # in a more efficient way.
```

Comment:
this is slow because it walks 1000s of files for every font

both schemes can be combined, for example `Yrsa-Italic[wght].ttf` and `Yrsa[wght].ttf`

some files have 2 dashes or more

```
ls /nix/store/zssva4si0q3jlz5ma8qkvm3b2w6c125c-google-fonts-unstable-2022-02-26//share/fonts/truetype/ | grep -E '^.*-.*-.*$'

PT_Sans-Caption-Web-Bold.ttf
PT_Sans-Caption-Web-Regular.ttf
PT_Sans-Narrow-Web-Bold.ttf
PT_Sans-Narrow-Web-Regular.ttf
PT_Sans-Web-BoldItalic.ttf
PT_Sans-Web-Bold.ttf
PT_Sans-Web-Italic.ttf
PT_Sans-Web-Regular.ttf
PT_Serif-Caption-Web-Italic.ttf
PT_Serif-Caption-Web-Regular.ttf
PT_Serif-Web-BoldItalic.ttf
PT_Serif-Web-Bold.ttf
PT_Serif-Web-Italic.ttf
PT_Serif-Web-Regular.ttf
```

all files

```
ls /nix/store/zssva4si0q3jlz5ma8qkvm3b2w6c125c-google-fonts-unstable-2022-02-26/share/fonts/truetype/ | wc -l

3313
```

regex for all files

```
find /nix/store/zssva4si0q3jlz5ma8qkvm3b2w6c125c-google-fonts-unstable-2022-02-26/share/fonts/truetype/ -regextype posix-extended -regex '.*/.*(-[A-Za-z]+)?(\[[a-z]+\])?.ttf' | wc -l

3313
```

fastest filter

```sh
escapeStringForRegex() {
  sed -E 's/[][|(){}.+*?\\]/\\&/g'
}
names_regex='('
for name in $names; do
  names_regex+=$(echo "$name" | escapeStringForRegex)'|'
done
names_regex=${names_regex:0: -1}')'
echo "names_regex=${names_regex@Q}" # debug

find . -regextype posix-extended -regex '.*/'"$names_regex"'(-[A-Za-z]+)?(\[[a-z]+\])?.ttf'
```

Reply:
but more generally:

google-fonts is a huge package (1.5 GB uncompressed)
and i would prefer one package per font family

let me have a look ...

Reply:
https://github.com/NixOS/nixpkgs/pull/215335